src/transformers/generation/logits_process.py

```diff
@@ -1702,3 +1702,28 @@ def __call__(self, input_ids, scores):
         unconditional_logits = torch.nn.functional.log_softmax(logits[:, -1], dim=-1)
         out = self.guidance_scale * (scores - unconditional_logits) + unconditional_logits
         return out
+
+
+class BarkEarlyStoppingLogitsProcessor(LogitsProcessor):
+    r"""This processor will set every tokens' log probability other than the EOS token to `-inf`
+    when the probabiliy of the EOS token id is superior to min_eos_p.
+
+    Args:
+        generate_config (`GenerateConfig`):
+            The generate config used to generate the output. The following parameters are required:
+                eos_token_id (`int`, *optional*, defaults to 50257):
+                    The id of the *end-of-sequence* token.
+                min_eos_p (`float`, *optional*, defaults to None):
```

Comment:
Nit, you don't need to precise that it defaults to `None`:
```suggestion
                min_eos_p (`float`, *optional*):
```